assert-on-string-literal message is wrong for empty literals

<!--
  Hi there! Thank you for discovering and submitting an issue.

  Before you submit this, make sure that the issue doesn't already exist
  or if it is not closed.

  Is your issue fixed on the preview release?: pip install pylint astroid --pre -U

-->

### Steps to reproduce
1. Install pylint from master
2. Run pylint on the following code:
```python
assert ""
```

### Current behavior
```
$ pylint test.py
test.py:1:0: W0129: Assert statement has a string literal as its first argument. The assert will never fail. (assert-on-string-literal)
```

### Expected behavior

This assert will fail, so the message should not state the contrary.

It's not clear to me if the message should be changed, or if assert-on-string-literal should be changed ignore empty strings.

### pylint --version output
```
pylint 2.5.0-dev1
astroid 2.4.0
Python 3.7.3 | packaged by conda-forge | (default, Jul  1 2019, 14:38:56) 
[Clang 4.0.1 (tags/RELEASE_401/final)]
```

--- a/pylint/checkers/base.py
+++ b/pylint/checkers/base.py
@@ -1017,7 +1017,7 @@ class BasicChecker(_BasicChecker):
             "intended to do.",
         ),
         "W0129": (
-            "Assert statement has a string literal as its first argument. The assert will never fail.",
+            "Assert statement has a string literal as its first argument. The assert will %s fail.",
             "assert-on-string-literal",
             "Used when an assert statement has a string literal as its first argument, which will "
             "cause the assert to always pass.",
@@ -1394,7 +1394,11 @@ class BasicChecker(_BasicChecker):
             self.add_message("assert-on-tuple", node=node)
 
         if isinstance(node.test, astroid.Const) and isinstance(node.test.value, str):
-            self.add_message("assert-on-string-literal", node=node)
+            if node.test.value:
+                when = "never"
+            else:
+                when = "always"
+            self.add_message("assert-on-string-literal", node=node, args=(when,))
 
     @utils.check_messages("duplicate-key")
     def visit_dict(self, node):

--- a/tests/functional/a/assert_on_string_literal.py
+++ b/tests/functional/a/assert_on_string_literal.py
@@ -1,3 +1,4 @@
 # pylint: disable=missing-module-docstring, undefined-variable
 assert [foo, bar], "No AssertionError"
 assert "There is an AssertionError" # [assert-on-string-literal]
+assert "" # [assert-on-string-literal]

--- a/tests/functional/a/assert_on_string_literal.txt
+++ b/tests/functional/a/assert_on_string_literal.txt
@@ -1,1 +1,2 @@
 assert-on-string-literal:3::Assert statement has a string literal as its first argument. The assert will never fail.
+assert-on-string-literal:4::Assert statement has a string literal as its first argument. The assert will always fail.

